src/org/rascalmpl/uri/URIResolverRegistry.java

```diff
@@ -120,6 +134,57 @@ public Set<String> getRegisteredClassloaderSchemes() {
 		return Collections.unmodifiableSet(classloaderResolvers.keySet());
 	}
 
+	private Object constructService(String name) throws ClassNotFoundException, InstantiationException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, SecurityException {
+		Class<?> clazz = Thread.currentThread().getContextClassLoader().loadClass(name);
+
+		try {
+			return clazz.getDeclaredConstructor(URIResolverRegistry.class).newInstance(this);
+		}
+		catch (NoSuchMethodException e) {
+			return clazz.newInstance();
+		}
+	}
+
+	private void loadFallback(String fallbackClass) {
+		try {
+			Object instance = constructService(fallbackClass);
+			boolean ok = false;
+			if (instance instanceof ILogicalSourceLocationResolver) {
+				fallbackLogicalResolver.set((ILogicalSourceLocationResolver) instance);
+				ok = true;
+			}
+
+			if (instance instanceof ISourceLocationInput) {
+				fallbackInputResolver.set((ISourceLocationInput) instance);
+				ok = true;
+			}
+
+			if (instance instanceof ISourceLocationOutput) {
+				fallbackOutputResolver.set((ISourceLocationOutput) instance);
+				ok = true;
+			}
+
+			if (instance instanceof IClassloaderLocationResolver) {
+				fallbackClassloaderResolver.set((IClassloaderLocationResolver) instance);
+				ok = true;
+			}
+
+			if (instance instanceof ISourceLocationWatcher) {
+				fallbackWatcher.set((ISourceLocationWatcher) instance);
+			}
+			if (!ok) {
+				System.err.println("WARNING: could not load fallback resolver " + fallbackClass
+					+ " because it does not implement ISourceLocationInput or ISourceLocationOutput or ILogicalSourceLocationResolver");
+			}
+		}
+		catch (ClassNotFoundException | InstantiationException | IllegalAccessException | ClassCastException
+			| IllegalArgumentException | InvocationTargetException | SecurityException  e) {
+			System.err.println("WARNING: could not load resolver due to " + e.getMessage());
+			e.printStackTrace();
```

Comment:
remove debug printStackTrace

Reply:
This message was copied from how normal errors are reported during loading of the other resolvers. I wouldn't classify it as a debug print.

Reply:
ok let's leave it here then.